src/nagini_translation/lib/program_nodes.py

```diff
@@ -343,6 +343,25 @@ def get_contents(self, only_top: bool) -> Dict:
     def python_class(self) -> 'PythonClass':
         return self.bound
 
+class SingletonFreshName:
```

Comment:
Would it be possible to move this class to util.py or something like that, since it's not a program node?

Reply:
Sure, done.